Removing all references of host-egress topic

tahmidefaz opened the pull request:
## Overview

This PR removes all references to the deprecated `platform.inventory.host-egress` topic.

## Secure Coding Practices Checklist GitHub Link

- https://github.com/RedHatInsights/secure-coding-checklist

## Secure Coding Checklist

- [x] General Coding Practices

kruai reviewed (changes requested):
If we're getting rid of the host-egress topic altogether, I think we also need to remove the references to `HOST_EGRESS_TOPIC`. They can all be changed to `EVENT_TOPIC` now, right?

tahmidefaz commented:
> If we're getting rid of the host-egress topic altogether, I think we also need to remove the references to `HOST_EGRESS_TOPIC`. They can all be changed to `EVENT_TOPIC` now, right?

Actually, we still use that environment variable in all our deployments: [here](https://gitlab.cee.redhat.com/service/app-interface/-/blob/master/data/services/insights/host-inventory/deploy.yml#L86)

kruai commented:
> > If we're getting rid of the host-egress topic altogether, I think we also need to remove the references to `HOST_EGRESS_TOPIC`. They can all be changed to `EVENT_TOPIC` now, right?
> 
> Actually, we still use that environment variable in all our deployments: [here](https://gitlab.cee.redhat.com/service/app-interface/-/blob/master/data/services/insights/host-inventory/deploy.yml#L86)

We do set the environment variable in the deployment templates, but we can remove it from the templates once we get rid of the refs in our code.

thearifismail commented on utils/kafka_consumer.py:
```diff
@@ -8,7 +8,7 @@
 
 
-HOST_EGRESS_TOPIC = os.environ.get("KAFKA_HOST_EGRESS_TOPIC", "platform.inventory.host-egress")
+HOST_EGRESS_TOPIC = os.environ.get("KAFKA_HOST_EGRESS_TOPIC", "platform.inventory.events")
```
do other teams/services use `KAFKA_HOST_EGRESS_TOPIC`?  Shouldn't this env var be named `KAFKA_HOST_EVENT_TOPIC`?

kruai replied:
We have a var named `KAFKA_EVENT_TOPIC` that we use for events

tahmidefaz replied:
Yeah, Asa had the same thought as well. Fixed it

thearifismail reviewed (dismissed):
asked a question about a minor issue for clarity or consitency

tahmidefaz commented:
> > > If we're getting rid of the host-egress topic altogether, I think we also need to remove the references to `HOST_EGRESS_TOPIC`. They can all be changed to `EVENT_TOPIC` now, right?
> > 
> > 
> > Actually, we still use that environment variable in all our deployments: [here](https://gitlab.cee.redhat.com/service/app-interface/-/blob/master/data/services/insights/host-inventory/deploy.yml#L86)
> 
> We do set the environment variable in the deployment templates, but we can remove it from the templates once we get rid of the refs in our code.

I see what you mean. Updated :+1:

kruai reviewed (changes requested):
Good progress, but there are still other references in the project to the host-egress topic. These should be changed to reference the events topic. Here are a few:
- https://github.com/RedHatInsights/insights-host-inventory/blob/master/app/__init__.py#L52
- https://github.com/RedHatInsights/insights-host-inventory/blob/master/app/config.py#L226
- https://github.com/RedHatInsights/insights-host-inventory/blob/master/app/queue/event_producer.py#L19
- https://github.com/RedHatInsights/insights-host-inventory/blob/master/tests/test_unit.py#L1690

tahmidefaz commented:
@kruai Just removed all refs of `host-egress`. Lmk, if you see anything that I missed.

I also remembered that we introduced an additional environment variable called `KAFKA_SECONDARY_TOPIC_ENABLED` during the transition period to `platform.inventory.events` topic. I went ahead and removed that as well since we don't need it anymore.

kruai reviewed (approved):
I think you got them all!

kruai reviewed (changes requested):
Oops sorry, there's actually one more reference to the secondary-topic-enabled value:
https://github.com/RedHatInsights/insights-host-inventory/blob/remove-host-egress-string/tests/fixtures/mq_fixtures.py#L75

tahmidefaz commented:
> Oops sorry, there's actually one more reference to the secondary-topic-enabled value:
> https://github.com/RedHatInsights/insights-host-inventory/blob/remove-host-egress-string/tests/fixtures/mq_fixtures.py#L75

I did remove that, but forgot to save the file :man_facepalming:

ryandillinfelton commented on app/queue/event_producer.py:
```diff
@@ -9,14 +9,14 @@
 logger = get_logger(__name__)
 
-Topic = Enum("Topic", ("egress", "events"))
+Topic = Enum("Topic", ("events"))
```
For all of the enums and things like this that are built around us having multiple egress topics it seems like we should switch them to just be singular since as far as I know there's no plan to have multiple again, and it doesn't seem all that likely that one will be made.

tahmidefaz replied:
@ryandillinfelton Good idea! I am working on it right now! If I push up my refactoring changes that you suggested here, it will make this PR quite large. Therefore, I am suggesting that we open a subsequent refactoring PR to address these changes. What's your opinion? (I am almost done, btw. Just trying to address a test failure)

ryandillinfelton reviewed (approved):
This all looks like it'll work just fine so I'll approve. However I do feel like while you're at it you might as well adjust some of the code to only expect one topic.

tahmidefaz commented:
As promised, here is the follow-up refactoring PR! https://github.com/RedHatInsights/insights-host-inventory/pull/840 :slightly_smiling_face: 
@ryandillinfelton
